From: Carlos Garnacho <carlosg@gnome.org>
Date: Fri, 16 Dec 2022 23:06:33 +0100
Subject: backends: Only apply EDID-based tablet mapping heuristic on
 integrated devices

These are the ones attached to a display, thus they are the ones that may need
help from this heuristic. Non-integrated tablets (e.g. Intuos) will default to
the span of all monitors.

Fixes mapping of opaque tablets if a display-integrated tablet of the same
brand is also plugged in.

(cherry picked from commit ba25271408a32a2a73a82acc6e094a611001c9f0)

Forwarded: https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/2767
Forwarded: https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/2763
Origin: upstream, 43.3, commit:933c9937f4fec0b54bf8e73033f9126f18d434e6
---
 src/backends/meta-input-mapper.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/backends/meta-input-mapper.c b/src/backends/meta-input-mapper.c
index 56da701..c5a9ee3 100644
--- a/src/backends/meta-input-mapper.c
+++ b/src/backends/meta-input-mapper.c
@@ -493,7 +493,7 @@ guess_candidates (MetaInputMapper     *mapper,
 
       g_assert (META_IS_MONITOR (l->data));
 
-      if (match_edid (input, l->data, &edid_match))
+      if (integrated && match_edid (input, l->data, &edid_match))
         match.score |= 1 << edid_match;
 
       if (integrated && match_size (input, l->data))
